main.go

```diff
@@ -26,24 +28,26 @@ func main() {
 	// Create a new Discord session using the provided bot token.
 	dg, err := discordgo.New("Bot " + Token)
 	if err != nil {
-		fmt.Println("error creating Discord session,", err)
+		log.Fatal("Error creating Discord session", err)
 		return
 	}
 
 	// Register the messageCreate func as a callback for MessageCreate events.
+	// Every time a message is created, func messaCreate is fired.
 	dg.AddHandler(messageCreate)
 
-	// In this example, we only care about receiving message events.
+	// Identify is sent during initial handshake with the discord gateway.
+	// In this case, will listen to group messages.
 	dg.Identify.Intents = discordgo.IntentsGuildMessages
 
 	// Open a websocket connection to Discord and begin listening.
 	err = dg.Open()
 	if err != nil {
-		fmt.Println("error opening connection,", err)
+		log.Fatal("error opening connection,", err)
 		return
```

Comment:
You don't need of `return` statement here. Cause, `log.Fatal` function emit a `os.Exit(1)`

Reply:
See: 
https://golang.org/src/log/log.go?s=10278:10306#L323

Reply:
Ok